### PR TITLE
deps: use native Promise instead of bluebird

### PR DESCRIPTION
**Pull request by omsmith**
Just extracted _promised-method_ from node-auth-validation (https://github.com/Brightspace/node-auth-validation/blob/master/src/promised.js).

There exists a similar package, but wasn't a fan of the implementation, even though it's all trivial.

### CODE REVIEW

File: src/abstract-cache.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ AbstractLandlordCache.prototype.getTenantIdLookup = Promise.method(/* @this */ f
 		});
 });
 
-AbstractLandlordCache.prototype.cacheTenantIdLookup = Promise.method(/* @this */ function cacheTenantIdLookup(host, tenantId) {
+AbstractLandlordCache.prototype.cacheTenantIdLookup = promised(/* @this */ function cacheTenantIdLookup(host, tenantId) {
```
Comment (ppaskaris-d2l) [inline src/abstract-cache.js]: There's a `.return` [later](https://github.com/omsmith/node-landlord-client/blob/da9b850e7e182f2077d43120d930234c773b900e/src/abstract-cache.js#L58) that's from Bluebird. [Here also](https://github.com/omsmith/node-landlord-client/blob/da9b850e7e182f2077d43120d930234c773b900e/src/abstract-cache.js#L1123). 

[API Documentation](http://bluebirdjs.com/docs/api/return.html)

File: src/index.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ function LandlordClient(opts) {
 	this._landlordOpts = url.parse(this._landlord);
 }
 
-LandlordClient.prototype.lookupTenantId = Promise.method(/* @this */ function lookupTenantId(host) {
+LandlordClient.prototype.lookupTenantId = promised(/* @this */ function lookupTenantId(host) {
```
Comment (ppaskaris-d2l) [inline src/index.js]: Bluebird `.return()` [in here](https://github.com/omsmith/node-landlord-client/blob/da9b850e7e182f2077d43120d930234c773b900e/src/index.js#L80) as well.

[And here](https://github.com/omsmith/node-landlord-client/blob/da9b850e7e182f2077d43120d930234c773b900e/src/index.js#L153)

**Comment by omsmith** [comment]
Native promises have been getting faster, and have debug/trace-ability benefits over bluebird.

**Comment by ppaskaris-d2l** [comment]
You were very quick with that amend. Ignore my review comments.

File: src/index.js (hunk 2 above)
Comment (omsmith) [inline src/index.js]: Yeah caught these ones after the first push.

File: src/abstract-cache.js (hunk 1 above)
Comment (omsmith) [inline src/abstract-cache.js]: Didn't catch these ones. Obviously the tests don't exercise this! Ha, better than before :)

**Comment by omsmith** [comment]
@ppaskaris-d2l should be good now

**Pull request merged**